src/plugins/blackbox/public/js/blackbox.js

```diff
@@ -616,10 +616,55 @@
       
     }.bind(null, options.collection));
   };
+
+  Blackbox.prototype._exportVideoToNativeFileSystem = function _exportVideoToNativeFileSystem(options,callback){
+    var initFrame;
+    var session_record;
+    var save=null;
+    var self=this;
+
+    return new Promise(function(resolve,reject){
+        window.OROVE.AltStorage.GetFileHandle({defaultPath:'mp4-' + options.sessionID + '.' + 'mp4'},function(err,callback){
```

Comment:
GetFileHandle invokes the native file save dialog and returns a callback that when invoked will append data to the flie that is created.